packages/design-system/package.json

```diff
@@ -0,0 +1,29 @@
+{
+  "name": "@repo/design-system",
+  "version": "0.0.0",
+  "exports": {
+    ".": {
+      "import": "./dist/index.mjs",
+      "types": "./index.ts"
```

Comment:
It is important that you list `types` before the other stuff like `import` and `require` in order for the types to fully work correctly.

Also you might want to think about adding in at the root level of this config the following (usually just above "exports"):

```json
"types": "./dist/index.d.ts",
"main": "./dist/index.mjs",
```

Make sure to use the right paths, above is just an example.